src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8573,7 +8567,7 @@ private VariableDesignationSyntax ParseDeconstructionDesignation(bool topLevel =
         /// </summary>
         private bool IsPossibleDeconstructionDeclaration()
         {
-            if ((this.CurrentToken.IsVar() && this.PeekToken(1).Kind == SyntaxKind.OpenParenToken) ||
+            if (((this.CurrentToken.IsVar() || IsPredefinedType(this.CurrentToken.Kind)) && this.PeekToken(1).Kind == SyntaxKind.OpenParenToken) ||
```

Comment:
Consider creating a helper for `token.IsVar() || IsPredefinedType(token.Kind)`. There are multiple uses of that expression here.